pkg/chains/formats/slsa/v2alpha2/internal/resolved_dependencies/resolved_dependencies_test.go

```diff
@@ -0,0 +1,511 @@
+/*
+Copyright 2023 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resolveddependencies
+
+import (
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/ghodss/yaml"
+	"github.com/google/go-cmp/cmp"
+	"github.com/in-toto/in-toto-golang/in_toto/slsa_provenance/common"
+	v1 "github.com/in-toto/in-toto-golang/in_toto/slsa_provenance/v1"
+	"github.com/tektoncd/chains/internal/backport"
+	"github.com/tektoncd/chains/pkg/artifacts"
+	"github.com/tektoncd/chains/pkg/chains/objects"
+	"github.com/tektoncd/chains/pkg/internal/objectloader"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"github.com/tektoncd/pipeline/pkg/apis/resource/v1alpha1"
+	logtesting "knative.dev/pkg/logging/testing"
+)
+
+const digest = "sha256:05f95b26ed10668b7183c1e2da98610e91372fa9f510046d4ce5812addad86b7"
+
+var pro *objects.PipelineRunObject
+var proStructuredResults *objects.PipelineRunObject
+
+func init() {
+	pro = createPro("../../../testdata/pipelinerun1.json")
+	proStructuredResults = createPro("../../../testdata/pipelinerun_structured_results.json")
+}
+
+func createPro(path string) *objects.PipelineRunObject {
+	var err error
+	pr, err := objectloader.PipelineRunFromFile(path)
+	if err != nil {
+		panic(err)
+	}
+	tr1, err := objectloader.TaskRunFromFile("../../../testdata/taskrun1.json")
+	if err != nil {
+		panic(err)
+	}
+	tr2, err := objectloader.TaskRunFromFile("../../../testdata/taskrun2.json")
+	if err != nil {
+		panic(err)
+	}
+	p := objects.NewPipelineRunObject(pr)
+	p.AppendTaskRun(tr1)
+	p.AppendTaskRun(tr2)
+	return p
+}
+
+func TestTaskRunWithTaskRunResults(t *testing.T) {
+	// make sure this works with Git resources
+	taskrun := `apiVersion: tekton.dev/v1beta1
+kind: TaskRun
+spec:
+  taskSpec:
+    resources:
+      inputs:
+      - name: repo
+        type: git
+status:
+  taskResults:
+  - name: CHAINS-GIT_COMMIT
+    value: 50c56a48cfb3a5a80fa36ed91c739bdac8381cbe
+  - name: CHAINS-GIT_URL
+    value: https://github.com/GoogleContainerTools/distroless`
```

Comment:
Is it possible to add this as a test case to TestTaskRun function?

Reply:
Actually, we already cover that test in TestTaskRun as well so I think I can just omit it. Even when I test without it, the coverage is unchanged.